src/lib/Form/DataMapper/UserRegisterMapper.php

```diff
@@ -57,23 +58,45 @@ public function mapToFormData()
         $this->configureOptions($resolver);
         $this->params = $resolver->resolve($this->params);
 
-        $contentType = $this->contentTypeLoader->loadContentType();
+        /** @var \Ibexa\Contracts\User\Invitation\Invitation|null $invitation */
+        $invitation = $this->params['invitation'] ?? null;
+        $contentType = $this->contentTypeLoader->loadContentType(
+            $invitation ? $invitation->getSiteAccessIdentifier() : null
+        );
 
         $data = new UserRegisterData([
             'contentType' => $contentType,
             'mainLanguageCode' => $this->params['language'],
             'enabled' => true,
         ]);
-        $data->addParentGroup($this->parentGroupLoader->loadGroup());
 
+        if ($invitation && $invitation->getUserGroup()) {
+            $targetGroup = $invitation->getUserGroup();
+        } else {
+            $targetGroup = $this->parentGroupLoader->loadGroup();
+        }
+
+        $data->addParentGroup($targetGroup);
+
+        if ($invitation) {
+            $data->setRole($invitation->getRole());
+            $data->setRoleLimitation($invitation->getLimitation());
```

Comment:
Static code analysis reports thatsetRole/setRoleLimitation doesn't exists

Reply:
It is in this related PR:
https://github.com/ibexa/content-forms/pull/29/files#diff-5b31a9d2b46819a6a6de21a9505594ecfca40f30ff0765105dd33bb4d97c31feR80